src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XContainer.cs

```diff
@@ -807,51 +807,31 @@ private IEnumerable<XElement> GetElements(XName? name)
 
         internal static string GetStringValue(object value)
         {
-            string? s = value as string;
-            if (s != null)
-            {
-                return s;
-            }
-            else if (value is double)
-            {
-                s = XmlConvert.ToString((double)value);
-            }
-            else if (value is float)
-            {
-                s = XmlConvert.ToString((float)value);
-            }
-            else if (value is decimal)
-            {
-                s = XmlConvert.ToString((decimal)value);
-            }
-            else if (value is bool)
-            {
-                s = XmlConvert.ToString((bool)value);
-            }
-            else if (value is DateTime)
-            {
-                s = XmlConvert.ToString((DateTime) value, XmlDateTimeSerializationMode.RoundtripKind);
-            }
-            else if (value is DateTimeOffset)
-            {
-                s = XmlConvert.ToString((DateTimeOffset)value);
-            }
-            else if (value is TimeSpan)
-            {
-                s = XmlConvert.ToString((TimeSpan)value);
-            }
-            else if (value is XObject)
-            {
-                throw new ArgumentException(SR.Argument_XObjectValue);
-            }
-            else
-            {
-                s = value.ToString();
-            }
+            string? s = ConvertToString(value);
             if (s == null) throw new ArgumentException(SR.Argument_ConvertToString);
             return s;
         }
 
+        private static string? ConvertToString(object value) =>
```

Comment:
Do we need to factor this out into separate method? I think it can be inline in the original method just fine.

Reply:
Yes it can be inline in the original method, just there is already many `if else` and i needed to add 4 more, is there any reason to keep the original?

Reply:
`GetStringValue` adds just a null check to the switch statement now (ie 1 or 2 extra lines). I do not see the benefit of keeping this null check in a separate method. I think having the null check in a separate method makes the code a bit harder to follow actually.

Reply:
Oh i see your point, moving the switch to the original method